BuildScript/BuildTasks/TestConfigurationsFactory.cs

```diff
@@ -144,11 +144,46 @@ private string GetPlatform (IEnumerable<string> configurationItems)
 
     private ExecutionRuntime GetExecutionRuntime (IEnumerable<string> configurationItems)
     {
-      var kvp = _supportedExecutionRuntimes.Single (configurationItems, () => CreateMissingConfigurationItemException ("execution runtime"));
+      var enumeratedConfigurationItems = configurationItems.ToArray();
+      if (TryGetDockerImageIfIsEnforcedLocalMachine (enumeratedConfigurationItems, out var dockerImageOfEnforcedLocalMachine))
```

Comment:
I'm not sure I understand the required complexity here. If we always add the docker image if there is one, could we then not simplify things?

Reply:
I actually thought doing it like this would reduce complexity. My approach was to not treat `EnforcedLocalMachine` **only** as a flag but additionally allow to provide a docker image along with it. This way I wouldn't have to change existing business logic and architecture and just add one extra branch to `GetExecutionRuntime`'s flow.

If we treated it only as a flag, we'd still need extra handling for `LocalMachine` which never has a docker image associated with it

Reply:
hmm...I'm not convinced this would hurt readability. probably need a pairsession?

Reply:
Just to be clear: you explicitly want `EnforcedLocalMachine` to only be an extra flag and not an execution runtime on its own like it was until know, did I understand that correctly?